azure-pipelines.yml

```diff
@@ -65,11 +65,61 @@ jobs:
               _Category: CoreCLR
               _Framework: netcoreapp3.0
               _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: '--benchview-submission-name "$(BenchviewCommitName)" --benchview-machinepool perfsnake --generate-benchview-data --benchview-run-type $(_RunType) $(UploadToBenchview)'
             x86_CoreCLR_netcoreapp30:
               _Architecture: x86
               _Category: CoreCLR
               _Framework: netcoreapp3.0
               _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: '--benchview-submission-name "$(BenchviewCommitName)" --benchview-machinepool perfsnake --generate-benchview-data --benchview-run-type $(_RunType) $(UploadToBenchview)'
+            x64_CoreCLR_netcoreapp22:
+              _Architecture: x64
+              _Category: CoreCLR
+              _Framework: netcoreapp2.2
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x86_CoreCLR_netcoreapp22:
+              _Architecture: x86
+              _Category: CoreCLR
+              _Framework: netcoreapp2.2
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x64_CoreCLR_netcoreapp21:
+              _Architecture: x64
+              _Category: CoreCLR
+              _Framework: netcoreapp2.1
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x86_CoreCLR_netcoreapp21:
+              _Architecture: x86
+              _Category: CoreCLR
+              _Framework: netcoreapp2.1
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x64_CoreCLR_netcoreapp20:
+              _Architecture: x64
+              _Category: CoreCLR
+              _Framework: netcoreapp2.0
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x86_CoreCLR_netcoreapp20:
+              _Architecture: x86
+              _Category: CoreCLR
+              _Framework: netcoreapp2.0
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x64_CoreCLR_full_framework:
+              _Architecture: x64
+              _Category: CoreCLR
+              _Framework: net461
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
+            x86_CoreCLR_full_framework:
+              _Architecture: x86
+              _Category: CoreCLR
+              _Framework: net461
+              _BuildConfig: $(_Architecture)_$(_Category)_$(_Framework)
+              _BenchViewArguments: ''
```

Comment:
@adiaaida I know that adding so many elements to the matrix looks a little bit lame, but I have no better idea. I am open to suggestions

Reply:
This is expected. I think it looks fine.

Only comment here is I think we should add some comments for like-blocks of definitions. So like above the netcoreapp3.0 ones, a comment saying # netcoreapp3.0 definitions; above the 2.2 # netcoreapp2.2, etc.